internal/controller/operatorconfigmap_controller.go

```diff
@@ -950,3 +987,59 @@ func (c *OperatorConfigMapReconciler) deleteDelegatedCSI() error {
 	}
 	return nil
 }
+
+func (c *OperatorConfigMapReconciler) getCsiCustomTolerations() ([]corev1.Toleration, []corev1.Toleration, error) {
+	nodePluginTolerations, controllerPluginTolerations := []corev1.Toleration{}, []corev1.Toleration{}
+
+	// list the rook ceph cm
+	rookCephOperatorConfig := &corev1.ConfigMap{}
+	err := c.Client.Get(c.ctx, types.NamespacedName{Name: utils.RookCephOperatorConfigName, Namespace: c.OperatorNamespace}, rookCephOperatorConfig)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			return nodePluginTolerations, controllerPluginTolerations, nil
+		}
+		return nodePluginTolerations, controllerPluginTolerations, fmt.Errorf("failed to get Rook Ceph Operator ConfigMap: %v", err)
+	}
+
+	// Get the CSI tolerations from the Rook Ceph Operator ConfigMap
+	csiPluginTolerations, plugin := rookCephOperatorConfig.Data[utils.CsiPluginTolerationKey]
+	csiProvisionerToleration, provisioner := rookCephOperatorConfig.Data[utils.CsiProvisionerTolerationKey]
+	if (!plugin && !provisioner) || (csiPluginTolerations == "" && csiProvisionerToleration == "") {
+		return nodePluginTolerations, controllerPluginTolerations, nil
+	}
+
+	if plugin && csiPluginTolerations != "" {
+		tolerations, err := getTolerations(csiPluginTolerations)
+		if err != nil {
+			return nodePluginTolerations, controllerPluginTolerations, fmt.Errorf("failed to parse csi plugin tolerations: %v", err)
+		}
+		nodePluginTolerations = tolerations
+	}
+
+	if provisioner && csiProvisionerToleration != "" {
+		tolerations, err := getTolerations(csiProvisionerToleration)
+		if err != nil {
+			return nodePluginTolerations, controllerPluginTolerations, fmt.Errorf("failed to parse csi provisioner tolerations: %v", err)
+		}
+		controllerPluginTolerations = tolerations
+	}
+
+	return nodePluginTolerations, controllerPluginTolerations, nil
+}
+
+func getTolerations(csiPluginTolerations string) ([]corev1.Toleration, error) {
```

Comment:
This function is about parsing the data not getting it
```suggestion
func parseTolerations(csiPluginTolerations string) ([]corev1.Toleration, error) {
```